### PR TITLE
adding helm charts to install karmada metrics adapter

### LINKED ISSUE #3626: feat: deploy karmada-metreics-adapter with helm/karmadactl/karmada-operator
**What would you like to be added**:
- [ ] Implement deployment of `karmada-metrics-adapter` using Helm. @sbdtu5498
- [x] Implement deployment of `karmada-metrics-adapter` using karmadactl. @chaunceyjiang 
- [x] Implement deployment of `karmada-metrics-adapter` using karmada-operator. @calvin0327

**Why is this needed**:
Provider different ways to install `karmada-metrics-adapter`, to utilize FederatedHPA and scale workloads across multiple clusters.  

If you are interested in one of the installation methods, you can choose to complete certain tasks.

### PR DESCRIPTION
**Pull request by sbdtu5498**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Adds helm charts to install karmada metrics adapter

**Which issue(s) this PR fixes**:
Fixes #3626 helm charts

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by jwcesign** [comment]
Hi, @sbdtu5498 Can u help fix the chart lint first?

**Comment by sbdtu5498** [comment]
Sure I will. Let me take a look.

File: charts/karmada/templates/_karmada_apiservice.tpl [hunk 1]
```diff
@@ -57,4 +57,60 @@ spec:
...
+---
+apiVersion: apiregistration.k8s.io/v1
+kind: APIService
+metadata:
+  name: v1beta1.metrics.k8s.io
+  labels:
+    app: {{ $name }}-metrics-adapter
+    apiserver: "true"
+spec:
+  insecureSkipTLSVerify: true
```
Comment (chaosi-zju) [inline charts/karmada/templates/_karmada_apiservice.tpl:70]: Hello, thank you for your important commit! 

While, we are trying to eliminate `insecureSkipTLSVerify=true` in karmada, detail in #4024

Can you refer to [karmada-metrics-adapter-apiservice.yaml](https://github.com/karmada-io/karmada/blob/master/artifacts/deploy/karmada-metrics-adapter-apiservice.yaml) and #4033 to remove `insecureSkipTLSVerify`?

Besides, could you please add steps of "how to verify your code" in this PR?

**Comment by calvin0327** [comment]
@sbdtu5498 Do you have a test for this? example: `helm template` or `helm install xxx --dry-run`

**Comment by jwcesign** [comment]
Hi, @sbdtu5498 Any updates?

**Comment by jwcesign** [comment]
Thanks for your contribution @sbdtu5498 , we need to release this feature at the end of this month.
I modified based on yours. let‘s follow this in https://github.com/karmada-io/karmada/pull/4303

**Pull request closed without merging**